ui/src/app/components/popups.tsx

```diff
@@ -66,14 +66,18 @@ const computeNewDimensions = (el, target) =>
 
 interface WithDynamicPositionProps {
   target: string;
+  handleClickOutside?: () => void;
+  outsideClickIgnoreClass?: string;
+  onClick?: () => void;
 }
 
 export const withDynamicPosition = () => (WrappedComponent) => {
-  const Wrapper = class WithDynamicPosition extends React.Component {
+  const Wrapper = class WithDynamicPosition extends React.Component<
```

Comment:
I am trying to find documentation about how React.Component working as a typed component. I was looking at the source, and I don't see how it accepts types. Do you have something documenting how this works?
https://github.com/facebook/react/blob/main/packages/react/src/ReactBaseClasses.js

Reply:
I do not. If you ask Chat GPT how to fix the type error that is thrown from the unchanged code, it will suggest the code as changed.

Reply:
Thanks for clarifying. That's helpful context.